test/langtools/jdk/javadoc/doclet/testDirectedInheritance/TestDirectedInheritance.java

```diff
@@ -0,0 +1,767 @@
+                }
+                """, """
+                package x;
+                public class D {
+                    /** @return {@inheritDoc MySuperType} */
+                    public Integer minus(Integer i) { return -i; }
+                }
+                """, """
+                package x;
+                public class E {
+                    /** @throws NullPointerException {@inheritDoc MySuperType} */
+                    public Integer minus(Integer i) { return -i; }
+                }
+                """, """
+                package x;
+                public class F {
+                    /** @throws T NullPointerException {@inheritDoc MySuperType} */
+                    public <T extends RuntimeException> Integer minus(Integer i) { return -i; }
+                }
+                """);
+        javadoc("-Xdoclint:none", // turn off DocLint
+                "-d", base.resolve("out").toString(),
+                "--source-path", src.toString(),
+                "x");
+        checkExit(Exit.ERROR);
+        new OutputChecker(Output.OUT).setExpectOrdered(false).check("""
+                A.java:3: error: cannot find the overridden method
+                    /** {@inheritDoc MySuperType} */
+                        ^""", """
+                B.java:3: error: cannot find the overridden method
+                    /** @param i {@inheritDoc MySuperType} */
+                                 ^""", """
+                C.java:3: error: cannot find the overridden method
+                    /** @param <T> {@inheritDoc MySuperType} */
+                                   ^""", """
+                D.java:3: error: cannot find the overridden method
+                    /** @return {@inheritDoc MySuperType} */
+                                ^""", """
+                E.java:3: error: cannot find the overridden method
+                    /** @throws NullPointerException {@inheritDoc MySuperType} */
+                                                     ^""", """
+                F.java:3: error: cannot find the overridden method
+                    /** @throws T NullPointerException {@inheritDoc MySuperType} */
+                                                       ^""");
+        new OutputChecker(Output.OUT).setExpectFound(false)
+                .check("warning: not a direct supertype"); // no unexpected warnings
+    }
+
+    /*
+     * Since a subtype refines contract of its supertypes, it does not seem
+     * immediately obvious that directed documentation inheritance should
+     * allow a method doc comment to inherit from that method type's
+     * indirect (not immediate) supertypes. This is similar to how an
+     * implementation cannot write, say, super.super.m().
+     *
+     *
+     * Directed documentation inheritance was primarily designed to
+     * resolve possible ambiguity among immediate supertypes.
+     *
+     * For now a warning is issued if a doc comment inherits from
+     * an indirect supertype.
```

Comment:
Not sure I agree with this. Can we discuss?

Reply:
Sure we can; this relates to an earlier comment of yours on Utils.isDirectSupertype here: https://github.com/openjdk/jdk/pull/14357#discussion_r1222053011

Reply:
The general criticism here is whether we should restrict in any way the set of super types from which one can inherit documentation, and/or what should the set of checks be?

For example, if a method in C inherits a method in B that has no comment but which inherits a method with a comment in A, then it seems unduly restrictive to stop the method in C explicitly referencing the method in A, as compared to only allowing a reference to B, which implicitly gets its comment from A.

That being said, there is merit in starting off with restrictions and loosening them in the face of experience, rather than trying to increase the stylistic restrictions later.

Reply:
> Sure we can; this relates to an earlier comment of yours on Utils.isDirectSupertype here: [#14357 (comment)](https://github.com/openjdk/jdk/pull/14357#discussion_r1222053011)

Thanks for tying these together; I had not realized they were related.

Reply:
As agreed out of band, I removed that check and the test for now in dea776e285b.